plugins/modules/system/facter.py

```diff
@@ -34,12 +51,19 @@
 
 def main():
     module = AnsibleModule(
-        argument_spec=dict()
+        argument_spec=dict(
+            arguments=dict(required=False, type='list', elements='str')
+        )
     )
 
-    facter_path = module.get_bin_path('facter', opt_dirs=['/opt/puppetlabs/bin'])
+    facter_path = module.get_bin_path(
+        'facter',
+        opt_dirs=['/opt/puppetlabs/bin'])
 
     cmd = [facter_path, "--json"]
+    if module.params['arguments']:
+        for argument in module.params['arguments']:
+            cmd += [argument.strip()]
```

Comment:
```suggestion
            cmd += [argument]
```
There's no reason why you want to mangle user input.

Also, why not just do:
```.py
if module.params['arguments']:
    cmd += module.params['arguments']
```
?

Reply:
switched it up to simpler format